src/webgl/p5.RendererGL.js

```diff
@@ -157,6 +143,20 @@ p5.RendererGL.prototype = Object.create(p5.Renderer.prototype);
 // Setting
 //////////////////////////////////////////////
 
+p5.RendererGL.prototype._setAttributeDefaults = function(attr) {
+  var attributes = attr || this._pInst._presetAttributes || {};
```

Comment:
Hi @stalgiag I can see a problem here, look at this case-

```js
function setup() {
  setAttributes('perPixelLighting', true);
  setAttributes('antialias', true);
  createCanvas(1000, 600, WEBGL);
}

function draw() {
    // code
}

function mousePressed() {
  setAttributes('perPixelLighting', false);
 }
```

Upon pressing the mouse, `antialias` will become `false`, as it reverts to its default value,

Probably using `Object.assign({}, attr, this._pInst._presetAttributes)` will correct the issue.

Reply:
Hi @sanketsingh24! You are right. Though that resetting to defaults on everything that isn't explicitly set behavior that you are describing is the current implementation but I agree that this isn't necessarily what people should expect. Alternatively we can always update the presetAttributes regardless of renderer and always use those. This will behave a little differently but is probably a more organized implementation.